### PR TITLE
Feat homepage UI

### PR DESCRIPTION
**Pull request by DreXuri**

### CODE REVIEW

**Comment by DreXuri** [comment]
@ikennaumeh

**Comment by DreXuri** [comment]
CC @moak13

File: lib/ui/views/homepage/homepage_view.dart [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                  crossAxisCount: 2,
+                  childAspectRatio: 0.67,
+                  mainAxisSpacing: 20.h,
+                  crossAxisSpacing: 16.w,
+                  mainAxisExtent: 252.h,
+                ),
+                itemCount: viewModel.productItems.length,
+                itemBuilder: (context, index) {
+                  return ProductItem(
+                    productModel: viewModel.productItems[index],
```
Comment (moak13) [inline lib/ui/views/homepage/homepage_view.dart:74]: kindly make use of the `elementAt` api for list items to get an element at a particular index. You can take a hint on this from the onboarding work that I did.

File: lib/ui/views/homepage/widgets/product_item.dart [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                  padding: EdgeInsets.all(12.0.h),
+                  child: SvgPicture.asset(AppImages.lkedIcon),
+                );
+              })
+            ],
+          ),
+          SizedBox(
+            height: 8.h,
+          ),
+          Row(
```
Comment (moak13) [inline lib/ui/views/homepage/widgets/product_item.dart:52]: Kindly make use of a `RichText` here.

File: lib/ui/views/homepage/widgets/product_item.dart [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+              Text(
+                productModel.title,
+                textAlign: TextAlign.center,
+                style: typography?.labelRegular12?.copyWith(
+                  color: palette?.primary6,
+                ),
+              ),
+            ],
+          ),
+          Row(
```
Comment (moak13) [inline lib/ui/views/homepage/widgets/product_item.dart:71]: Same as above

**Review by moak13 — CHANGES_REQUESTED** [review]
Awesome work, Kindly take note of the below.

File: lib/ui/views/homepage/homepage_view.dart (hunk 1 above)
Comment (DreXuri) [reply lib/ui/views/homepage/homepage_view.dart]: really what the different. I am sure this should not cause a break in code

File: lib/ui/views/homepage/homepage_view.dart [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                  ),
+                  child: SvgPicture.asset(
+                    AppImages.burgerLog,
+                    width: 24.7.w,
+                    height: 24.h,
+                  ),
+                ),
+              )),
+          title: Text(
+            'Home',
```
Comment (moak13) [inline lib/ui/views/homepage/homepage_view.dart:49]: move this to intl if your flutter version supports

File: lib/ui/views/homepage/homepage_view.dart [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                    crossAxisCount: 2,
+                    childAspectRatio: 0.67,
+                    mainAxisSpacing: 20.h,
+                    crossAxisSpacing: 16.w,
+                    mainAxisExtent: 252.h,
+                  ),
+                  itemCount: viewModel.productItems.length,
+                  itemBuilder: (context, index) {
+                    return ProductItem(
+                      productModel: viewModel.productItems.elementAt(index),
```
Comment (moak13) [inline lib/ui/views/homepage/homepage_view.dart:72]: not what i meant tho, pretty sure mine doesn't look like this

File: lib/ui/views/homepage/widgets/product_item.dart [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                padding: EdgeInsets.all(8.0.w),
+                child: Container(
+                  height: 32.h,
+                  width: 32.w,
+                  decoration: BoxDecoration(
+                    color: kcIconBackground,
+                    borderRadius: BorderRadius.circular(4.r),
+                  ),
+                  child: Builder(builder: (context) {
+                    if (productModel.isLiked) {
```
Comment (moak13) [inline lib/ui/views/homepage/widgets/product_item.dart:48]: not what i meant as well but it's fine.

**Review by moak13 — CHANGES_REQUESTED** [review]
Awesome work, check again

File: lib/ui/views/homepage/homepage_view.dart (hunk 4 above)
Comment (DreXuri) [reply lib/ui/views/homepage/homepage_view.dart:49]: it doesn't

File: lib/ui/views/homepage/homepage_view.dart (hunk 5 above)
Comment (DreXuri) [reply lib/ui/views/homepage/homepage_view.dart]: lol, closed

File: lib/ui/views/homepage/widgets/product_item.dart (hunk 6 above)
Comment (DreXuri) [reply lib/ui/views/homepage/widgets/product_item.dart]: okay , closed

File: lib/ui/views/homepage/homepage_view.dart (hunk 4 above)
Comment (moak13) [reply lib/ui/views/homepage/homepage_view.dart:49]: aiit

**Pull request merged**